### ISSUE TITLE
Cannot convert HTseq tagged SAM files to BAM

### ISSUE THREAD

**Issue by aleighbrown** [htseq/htseq#35]
Hello, 

I'm trying use the XF tag generated by HTseq for some downstream analysis. I'm easily able to write out to sam

```
python -m HTSeq.scripts.count downsampled.bam \
/SAN/vyplab/vyplab_reference_genomes/annotation/human/GRCh38/gencode.v34.annotation.gff3 \
--stranded yes --samout downsampled.tagged.sam 
```

But my problem comes when I try to convert the sam to a bam

```
samtools view downsampled.tagged.sam  > test.bam
[E::sam_parse1] missing SAM header
[W::sam_read1] Parse error at line 1
[main_samview] truncated file.
```

I wasn't able to find any google answers on it, nor if anyone else has come across this particular error . 

Caveat: for speed of testing I've been running HTSeq on downsampled BAMs. Possible the error comes from the downsampling and not an issue with HTSeq's sam output.

**Comment by iosonofabio** [comment]
Please attach <5MB input BAM and GTF files and describe version, OS, Python version

**Comment by aleighbrown** [comment]
OS: CentOS Linux 7 (Core)
Python: 3.8.5
HTSeq: 0.13.5

The GFF is just the one from Gencode, it would definitely be too large to attach so link for download is here:
http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_34/gencode.v34.annotation.gff3.gz

BAM attached here (just renamed bam.zip so that github will accept the upload, not actually zipped)
[downsampled.bam.zip](https://github.com/htseq/htseq/files/7458964/downsampled.bam.zip)

**Comment by iosonofabio** [comment]
Thank you.

I can't efficiently debug without a small GTF file. Please reproduce the bug with a small GTF file <5MB) and then I'll look at it.

If you can't do that, I'm afraid I'll have to close the issue: unfortunately there are a few higher priority items on my list than wading through gigabytes of annotations.

Thanks,
Fabio

**Comment by aleighbrown** [comment]
Expected output from this gff3

```
python -m HTSeq.scripts.count downsampled.bam \
> downsampled.gff3 \
> --stranded yes --samout downsampled.htseq.sam 
61 GFF lines processed.
Warning: Read A00420:310:HWVTKDMXX:2:1460:29026:2566 claims to have an aligned mate which could not be found in an adjacent line.
Warning: 15519 reads with missing mate encountered.
16077 alignment pairs processed.
ENSG00000126767.18	5561
ENSG00000234359.3	0
__no_feature	9640
__ambiguous	0
__too_low_aQual	101
__not_aligned	109
__alignment_not_unique	666
```

And then with samtools

```
samtools view elk1_1_8h.htseq.sam > test.bam
[E::sam_parse1] missing SAM header
[W::sam_read1] Parse error at line 1
[main_samview] truncated file.
```


[downsampled.gff3.zip](https://github.com/htseq/htseq/files/7461100/downsampled.gff3.zip)

**Comment by iosonofabio** [comment]
Thank you. I can confirm this is a bug due to a corner case whereby you want a templated SAM file, whereas the current release only deals with nontemplated SAM or templated, binary BAM outputs.

This is now fixed in commit cc9b101, which will be in the next release.

If you need this urgently, there are two workarounds:

1. Install from the github master branch
2. Use a BAM file as output instead of a SAM file. IF you really need the uncompressed text SAM file as output, convert it to SAM post-facto using samtools.

Notice that your BAM file is probably sorted by position, but you're not using the `-r pos` option. That will most likely result in bogus output.

Closing and thanks again for reporting.

**Comment by iosonofabio** [comment]
Tagging #34 as we need to merge this commit into that PR, and that process might cause a little bit of pain.
